spec/models/king_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe King, type: :model do
+  
+  describe "King's valid_move" do
+  	it "should be inside destination, blocked" do
+      game = create :game
+      king = game.pieces.find_by(rank: 1, file: 5, color: "white")
+      king2 = game.pieces.find_by(rank: 8, file: 5, color: "black")
+  		expect(king.valid_move?(1, 6)).to eq(false)
+  		expect(king.valid_move?(1, 4)).to eq(false)
+  		expect(king.valid_move?(2, 5)).to eq(false)
+      expect(king.valid_move?(2, 4)).to eq(false)
+      expect(king.valid_move?(2, 6)).to eq(false)
+      expect(king2.valid_move?(7, 5)).to eq(false)
+      expect(king2.valid_move?(7, 4)).to eq(false)
+      expect(king2.valid_move?(7, 6)).to eq(false)
+  	end
+
+    it "shouldn't be outside destination" do
```

Comment:
Right here, my suggestion is `"should properly evaluate moves that are not allowed for a king but are otherwise valid"`.